### DISCUSSION TITLE
Complete DS4 features

### DISCUSSION THREAD

**Discussion by shazibjeahngir** [LizardByte/community#225] [Sunshine] [answered]
if there was ONE thing that would make me uninstall parsec and only use sunshine+moonlight it would be proper touchpad/gyro abilities because a lot of games need both (especially wii and switch games).  We have access to DS4 inputs, but why not the gyro and touchpad?

**Comment by ReenigneArcher** [comment]
I'd probably move this to Moonlight's request board... since they only try to emulate an Xbox 360 pad right now, Sunshine only accepts those equivalents.

**Reply by ld892012** [reply]
I have already created a feature request like this,  just in case. ^^
[Sunshine](https://github.com/LizardByte/Sunshine/discussions/979)
[Moonlight](https://github.com/moonlight-stream/moonlight-qt/issues/960)

**Comment by ld892012** [comment]
Now Moonlight can support gyro and touchpad in this [Commit](https://github.com/moonlight-stream/moonlight-qt/commit/9af58af5e4aa536083ad62997135c1f9c03430ec) ! 👍

**Comment by Arbitrate3280** [comment]
https://github.com/LizardByte/Sunshine/pull/1396

look like this is going to be a reality, really hope they manage to complete linux support for this

**Comment by cgutman** [comment]
Support for the full set of DS4 features is available for testing now. Please give it a try and let me know how it performs. You'll need to use dev builds of both Moonlight and Sunshine listed below.

~~Moonlight Windows: https://ci.appveyor.com/api/buildjobs/lm0v4h2nc8ehy5xe/artifacts/MoonlightPortable-x64-r2079.zip~~
~~Moonlight macOS: https://ci.appveyor.com/api/buildjobs/ha66wtquq7u4qkf1/artifacts/Moonlight-r2079.dmg~~
~~Sunshine Windows (need to be signed in on GitHub to download): https://github.com/LizardByte/Sunshine/suites/14159253569/artifacts/793377489~~

**See below for updated links**

**Reply by danaigus** [reply]
Using a dualsense, I found a problem that makes the triggers not work with analog data, but only active or inactive like a digital button. In the stable version of moonlight-qt this does not happen.
Another minor problem is that when I'm using the bluetooth connection and I try to connect the usb cable in the dualsense, the control is duplicated in the host machine. Removing the usb cable returns to only one controller being recognized.
Windows host and client.

**Reply by cgutman** [reply]
The trigger issue was a bug in the version of SDL included in that build of Moonlight Qt. It is fixed in the builds linked here.

Here are new builds of Moonlight and Sunshine for testing:
Moonlight Windows: https://ci.appveyor.com/api/buildjobs/86obf0nwmskllw7l/artifacts/MoonlightPortable-x64-r2084.zip
Moonlight macOS: https://ci.appveyor.com/api/buildjobs/f4q4opwayiypl4wx/artifacts/Moonlight-r2084.dmg
Sunshine Windows (need to be signed in on GitHub to download): https://github.com/LizardByte/Sunshine/suites/14167610325/artifacts/794048974

**Comment by cgutman** [comment, marked as answer]
Support for the full set of DS4 features on Windows has been merged into [nightly builds](https://github.com/LizardByte/Sunshine/releases/nightly-dev). This includes motion sensors, RGB LED control, touchpad, and battery reporting. You'll also need nightly versions of the Moonlight clients until the next official release.

[a comment by an automated account is not shown]

**Comment by Arbitrate3280** [comment]
Has this been added for linux as well?

**Reply by ReenigneArcher** [reply]
I believe so, not 100% sure. Windows get's most of the attention these days.

**Comment by SFurnace** [comment]
Can I use Swith Pro Controller with gyro support?

**Reply by ReenigneArcher** [reply]
As of the latest beta, Nintendo Switch gamepad emulation is possible in Linux.

**Comment by MikeDrewitt** [comment]
Far as I can tell, DS4 (and therefor gyro) isn't support when using linux as a host. Do we know what's preventing this from working? Is there any plans to fix this moving forward?

**Reply by ReenigneArcher** [reply]
Sunshine supports DS5 in the latest beta on Linux.

**Comment by mrartifact** [comment]
Hello.


I want to use steam deck as a host and connect to it via android and control my ps4 gamepad with gyroscope. 


On version sunshine 2025.122.141614 my connection is recognized by steam input system as xbox gamepad and gyroscope setting is not available. I have "allow use of gamepad motion sensors" checked in moonlight settings.


I also tried to install sunshine on windows and there the input system of steam correctly recognizes my gamepad and transmits data from the gyroscope. At the same time, if you compare sunshine settings, you can see that the windows version has additional gamepad settings, which are not in the steam deck version.


Can you please help me how to set up gyro data transfer from gamepad via android to steam deck?
